From fc8180d16df90615f13f6eb03e4867843c62a0f5 Mon Sep 17 00:00:00 2001
From: Ian McLean <17928-imclean@users.noreply.drupalcode.org>
Date: Mon, 19 Feb 2024 22:26:44 +0000
Subject: [PATCH] Update 3 files

- /src/Plugin/Field/FieldFormatter/PhotoswipeResponsiveFieldFormatter.php
- /src/Plugin/Field/FieldFormatter/PhotoswipeFieldFormatter.php
- /photoswipe.module
---
 photoswipe.module                                           | 1 +
 .../Field/FieldFormatter/PhotoswipeFieldFormatter.php       | 6 ++++--
 .../FieldFormatter/PhotoswipeResponsiveFieldFormatter.php   | 1 +
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/photoswipe.module b/photoswipe.module
index 3388f43..a976df5 100644
--- a/photoswipe.module
+++ b/photoswipe.module
@@ -90,5 +90,6 @@ function photoswipe_field_migration_field_formatter_info() {
     'image' => ['photoswipe_image_formatter' => 'photoswipe_field_formatter'],
     'media_image' => ['photoswipe_image_formatter' => 'photoswipe_field_formatter'],
     'entity_reference' => ['photoswipe_image_formatter' => 'photoswipe_field_formatter'],
+    'entity_reference_entity_modify' => ['photoswipe_image_formatter' => 'photoswipe_field_formatter'],
   ];
 }
diff --git a/src/Plugin/Field/FieldFormatter/PhotoswipeFieldFormatter.php b/src/Plugin/Field/FieldFormatter/PhotoswipeFieldFormatter.php
index 840b1c0..ecc6ec2 100644
--- a/src/Plugin/Field/FieldFormatter/PhotoswipeFieldFormatter.php
+++ b/src/Plugin/Field/FieldFormatter/PhotoswipeFieldFormatter.php
@@ -23,6 +23,7 @@ use Symfony\Component\DependencyInjection\ContainerInterface;
  *   label = @Translation("Photoswipe"),
  *   field_types = {
  *     "entity_reference",
+ *     "entity_reference_entity_modify",
  *     "image"
  *   }
  * )
@@ -229,7 +230,8 @@ class PhotoswipeFieldFormatter extends FormatterBase {
    *   The modified settings form structure of this formatter.
    */
   public function addEntityReferenceSettings(array $element) {
-    if ($this->fieldDefinition->getType() !== 'entity_reference') {
+    if ($this->fieldDefinition->getType() !== 'entity_reference'
+    && $this->fieldDefinition->getType() !== 'entity_reference_entity_modify') {
       return $element;
     }
 
@@ -436,7 +438,7 @@ class PhotoswipeFieldFormatter extends FormatterBase {
   public static function isApplicable(FieldDefinitionInterface $field_definition) {
     $field_type = $field_definition->getType();
     if ($field_type === 'image' ||
-       ($field_type === 'entity_reference' && $field_definition->getSetting('target_type') === 'media')) {
+    (($field_type === 'entity_reference' || $field_type === 'entity_reference_entity_modify') && $field_definition->getSetting('target_type') === 'media')) {
       return TRUE;
     }
     return FALSE;
diff --git a/src/Plugin/Field/FieldFormatter/PhotoswipeResponsiveFieldFormatter.php b/src/Plugin/Field/FieldFormatter/PhotoswipeResponsiveFieldFormatter.php
index 0b79244..a952601 100644
--- a/src/Plugin/Field/FieldFormatter/PhotoswipeResponsiveFieldFormatter.php
+++ b/src/Plugin/Field/FieldFormatter/PhotoswipeResponsiveFieldFormatter.php
@@ -17,6 +17,7 @@ use Symfony\Component\DependencyInjection\ContainerInterface;
  *   label = @Translation("Photoswipe Responsive"),
  *   field_types = {
  *     "entity_reference",
+ *     "entity_reference_entity_modify",
  *     "image"
  *   }
  * )
-- 
GitLab

